app/src/main/java/com/techcourse/DispatcherServlet.java

```diff
@@ -1,51 +1,79 @@
 package com.techcourse;
 
+import com.interface21.webmvc.servlet.ModelAndView;
+import com.interface21.webmvc.servlet.View;
+import com.interface21.webmvc.servlet.mvc.HandlerAdapter;
+import com.interface21.webmvc.servlet.mvc.asis.ControllerHandlerAdapter;
+import com.interface21.webmvc.servlet.mvc.tobe.AnnotationHandlerAdapter;
+import com.interface21.webmvc.servlet.mvc.tobe.AnnotationHandlerMapping;
 import jakarta.servlet.ServletException;
 import jakarta.servlet.http.HttpServlet;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
+import java.util.Optional;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import com.interface21.webmvc.servlet.view.JspView;
 
 public class DispatcherServlet extends HttpServlet {
 
     private static final long serialVersionUID = 1L;
     private static final Logger log = LoggerFactory.getLogger(DispatcherServlet.class);
 
-    private ManualHandlerMapping manualHandlerMapping;
+    private HandlerMappingRegistry handlerMappingRegistry;
+    private HandlerAdapterRegistry handlerAdapterRegistry;
 
     public DispatcherServlet() {
     }
 
     @Override
     public void init() {
-        manualHandlerMapping = new ManualHandlerMapping();
+        handlerMappingRegistry = new HandlerMappingRegistry();
+        final ManualHandlerMapping manualHandlerMapping = new ManualHandlerMapping();
         manualHandlerMapping.initialize();
+        handlerMappingRegistry.addHandlerMapping(manualHandlerMapping);
+
+        final String basePackage = Application.class.getPackage().getName();
+        final AnnotationHandlerMapping annotationHandlerMapping = new AnnotationHandlerMapping(basePackage);
+        annotationHandlerMapping.initialize();
+        handlerMappingRegistry.addHandlerMapping(annotationHandlerMapping);
+
+        handlerAdapterRegistry = new HandlerAdapterRegistry();
+        handlerAdapterRegistry.addHandlerAdapter(new ControllerHandlerAdapter());
+        handlerAdapterRegistry.addHandlerAdapter(new AnnotationHandlerAdapter());
```

Comment:
각각 registry내로 책임 분배하는 건 어떤가요?
registry를 생성했다면, 각 mapping or adapter는 registry의 책임이 되는게 자연스러워 보입니다!

예를 들어, handlermapping or adapter를 수정해야하는데 dispatcherServlet의 클래스를 들여다보지 않아도 될 것 같아요!!
```suggestion
        handlerMappingRegistry = new HandlerMappingRegistry();
        handlerMappingRegistry.initialize()

        handlerAdapterRegistry = new HandlerAdapterRegistry();
        handlerAdapterRegistry.initialize()
```

Reply:
registry가 관련 클래스 생성을 담당하면 책임 분배가 좋아지겠네요. 👍 
제안 감사합니다!